doc/reference/plutus/reference-script-example.md

```diff
@@ -0,0 +1,148 @@
+
+- Collateral tx input(s) - these are provided and are forfeited in the event the Plutus script fails to execute.
+- A Plutus tx output with accompanying datum hash. This is the tx output that sits at the Plutus script address. It must have a datum hash, otherwise, it is unspendable.
+- The reference transaction input containing the corresponding Plutus script. We must create the transaction output containing the reference Plutus script.
+
+#### Creating the `Required Redeemer` Plutus spending script
+
+The plutus-example executable will automagically generate several Plutus scripts in the CLI-compatible text envelope format.
+
+Run the following commands:
+
+```bash
+git clone git@github.com:input-output-hk/plutus-apps.git
+
+cd plutus-apps/plutus-example
+
+cabal run exe:plutus-example
+```
+
+This will output `required-redeemer.plutus` in the `generated-plutus-scripts/v2` dir.
+
+#### Setting up a local Babbage node cluster
+
+There is a convenient script that will set up a Babbage cluster immediately on your local machine.
+
+Run the following command:
+
+```bash
+cabal install cardano-cli
+cabal install cardano-node
+./scripts/babbage/mkfiles.sh
+```
+
+To start your babbage cluster, you need to run the `example/run/all.sh` shell script.
+The remainder of this guide provides a brief walkthrough of the [shell script example](scripts/plutus/example-reference-script-usage.sh) that automatically creates a reference script and spends the utxo at
+the reference script's corresponding script address.
+
+#### Creating a reference script at a transaction output and
+#### sending ada to the script address (with a datum)
+
+In order to use a reference script, we must first create this script at a particular transaction output.
+
+```bash
+cardano-cli transaction build \
+  --babbage-era \
+  --cardano-mode \
+  --testnet-magic 42 \
+  --change-address "$utxoaddr" \
+  --tx-in "$txin" \
+  --tx-out "$utxoaddr+$lovelace" \
+  --tx-out "$plutusscriptaddr+$lovelace" \
+  --tx-out-datum-hash "$scriptdatumhash" \
+  --tx-out "$dummyaddress+$lovelaceattxindiv3" \
+  --tx-out-reference-script-file "$plutusscriptinuse" \
+  --protocol-params-file "$WORK/pparams.json" \
+  --out-file "$WORK/create-datum-output.body"
+```
+
+The following should be noted about this build command:
+
+Firstly, we are sending ada to the plutus script address along with a datum hash. This is reflected in the following lines:
+
+```bash
+...
+--tx-out "$plutusscriptaddr+$lovelace" \
+--tx-out-datum-hash "$scriptdatumhash" \
+...
+```
+
+We have seen this before in the [plutus-spending-script-example.md](doc/reference/plutus/plutus-spending-script-example.md).
+
+Secondly, we are creating a reference script at a tx output, which cannot be the same as the reference script address of our choice:
```

Comment:
> which cannot be the same as the reference script address of our choice`

It can. We could say "there is no restriction on which address we choose." or "a reference script can be guarded by any type of address and there's no restriction on it being referenced."

Reply:
I'll just remove it :+1: